File: monsoon3.0/src/main/java/wtf/monsoon/impl/module/movement/speed/VerusSpeed.java
package wtf.monsoon.impl.module.movement.speed;

import io.github.nevalackin.homoBus.Listener;
import io.github.nevalackin.homoBus.annotations.EventLink;
import wtf.monsoon.api.module.Module;
import wtf.monsoon.api.setting.ModeProcessor;
import wtf.monsoon.impl.event.EventPreMotion;

public class VerusSpeed extends ModeProcessor {
    public VerusSpeed(Module parentModule) {
        super(parentModule);
    }

    @EventLink
    public final Listener<EventPreMotion> eventPreMotionListener = e -> {
        if(player.isMoving()) {
            if(player.isOnGround()) {
                if(!mc.gameSettings.keyBindJump.isKeyDown()) {
                    player.setSpeed(player.getBaseMoveSpeed());
                    player.jump();
                }
            } else {
                player.setSpeed(player.getSpeed() * 1.03);
            }
        }
    };


}

File: monsoon-3.x-v2/src/main/java/wtf/monsoon/client/modules/movement/speed/VerusSpeed.kt
package wtf.monsoon.client.modules.movement.speed

import me.bush.eventbuskotlin.EventListener
import me.bush.eventbuskotlin.listener
import wtf.monsoon.backend.module.MulticlassModule
import wtf.monsoon.backend.module.mode.ModuleMode
import wtf.monsoon.client.event.EventMove
import wtf.monsoon.client.event.EventPacket
import wtf.monsoon.client.event.EventPreMotion
import wtf.monsoon.client.modules.movement.Speed

class VerusSpeed(name: String, parent: MulticlassModule) : ModuleMode<Speed>(name, parent) {

    @EventListener
    val preMotion = fun(e: EventMove) {
        if (player.moving) {
            if (player.isOnGround) {
                if (!mc.gameSettings.keyBindJump.isKeyDown) {
                    player.setSpeed(e, player.baseMoveSpeed)
                    player.jump()
                }
            } else {
                player.setSpeed(e, player.speed * 1.03)
            }
        }
    }

}